adserver/admin.py

```diff
@@ -65,9 +73,74 @@ class AdvertiserAdmin(RemoveDeleteMixin, admin.ModelAdmin):
 
     """Django admin configuration for advertisers."""
 
-    list_display = ("name", "report")
+    actions = ["action_create_draft_invoice"]
+    list_display = ("name", "report", "stripe_customer")
     prepopulated_fields = {"slug": ("name",)}
-    readonly_fields = ("modified", "created")
+    readonly_fields = ("modified", "created", "stripe_customer")
+
+    def action_create_draft_invoice(self, request, queryset):
+        """Create a draft invoice for this customer with metadata attached."""
+        if not settings.STRIPE_SECRET_KEY:
+            messages.add_message(
+                request,
+                messages.ERROR,
+                _("Stripe is not configured. Please set settings.STRIPE_SECRET_KEY."),
+            )
+            return
+
+        flight_start = timezone.now()
+        flight_end = flight_start + timedelta(days=30)
```

Comment:
Should we not put this on the actual Flight object, where we could then pull the billing data?

Reply:
Actually that's a great idea. It doesn't have to be either/or. It could be both and then we'll see what actually gets used by a real workflow. Ideally, each flight is an invoiceitem and the whole thing is an invoice.

Reply:
My thought process behind both is that as we move from billing for specific flights to topping up an account, we might want it on the advertiser. I'm happy to get both and remove one in the future.